rdflib/graph.py

```diff
@@ -617,18 +617,27 @@ def set(self, triple):
 
     def subjects(self, predicate=None, object=None):
         """A generator of subjects with the given predicate and object"""
+        subs = []
```

Comment:
Use sets for keeping track of seen (lists have O(n) lookup time).
```suggestion
        subs = set()
```

(And subsequently use `subs.add(s)`).

Reply:
Oh yes, this makes sense. I'll update this. Are any other changes required?

Reply:
@niklasl I have pushed the suggested changes.